examples/http-service-data-binding/http_service_data_binding.md

```diff
@@ -1,6 +1,6 @@
 # REST service - Payload data binding
 
-HTTP service payload data binding allows accessing the request payload using a resource signature parameter. The resource parameter which is bound to the request payload should be annotated with `@http:Payload` annotation. The resource parameter type should be `anydata`. If the data binding fails, a `400 Bad Request` response is sent to the client. Use this to access the request payload directly from the resource.
+HTTP service payload data binding allows accessing the request payload using a resource signature parameter. The resource parameter type should be of `anydata` subtype. The default payload parameter types are basic structured types such as `map`, `array`, `tuple`, `table`, `record` and `xml` which is a sequence type, which can be defined in the signature without any annotation. For other types, the `@http:Payload` annotation is required. If the signature includes more than one structured type, the `@http:Payload` annotation should be used to resolve any ambiguity. The default payload parameter support is limited to `POST`, `PUT`, `PATCH`, `DELETE`, and `DEFAULT` accessors. If the data binding process fails, the client receives a 400 Bad Request response. This feature allows direct access to the request payload from the resource.
```

Comment:
HTTP service payload data binding allows accessing the request payload using a resource signature parameter. The resource parameter type should be of `anydata` subtype. By default parameters with type `map `, `array`, `tuple`, `table`, `record` and `xml` are mapped to the payload. For other types, the `@http:Payload` annotation is required and If the signature includes more than one of the aforementioned types, the `@http:Payload` should be used to resolve the ambiguity. This behaviour is limited to `POST`, `PUT`, `PATCH`, `DELETE`, and `DEFAULT` accessors.  If the data binding process fails, the client receives a 400 Bad Request response. This feature allows direct access to the request payload from the resource.
```suggestion
HTTP service payload data binding allows accessing the request payload using a resource signature parameter. The resource parameter type should be of `anydata` subtype. By default parameters with type `map `, `array`, `tuple`, `table`, `record` and `xml` are mapped to the payload. For other types, the `@http:Payload` annotation is required and If the signature includes more than one of the aforementioned types, the `@http:Payload` should be used to resolve the ambiguity. This behaviour is limited to `POST`, `PUT`, `PATCH`, `DELETE`, and `DEFAULT` accessors.  If the data binding process fails, the client receives a 400 Bad Request response. This feature allows direct access to the request payload from the resource.
```

Reply:
Fixed via [fcd7eff](https://github.com/ballerina-platform/ballerina-distribution/pull/4119/commits/fcd7eff8fab4660b16bfcb622b4bbf9504761107)